src/frequenz/client/microgrid/_component_states.py

```diff
@@ -97,3 +192,64 @@ def from_pb(cls, evc_state: ev_charger.ComponentState) -> Self:
             return cls(cls.UNKNOWN)
 
         return cls(evc_state)
+
+
+class InverterComponentState(Enum):
+    """Component states of an inverter."""
+
+    UNSPECIFIED = inverter.ComponentState.COMPONENT_STATE_UNSPECIFIED
+    """Unspecified component state."""
+
+    OFF = inverter.ComponentState.COMPONENT_STATE_OFF
+    """Inverter is switched off."""
+
+    SWITCHING_ON = inverter.ComponentState.COMPONENT_STATE_SWITCHING_ON
+    """The inverter is starting up and needs some time to become fully operational."""
+
+    SWITCHING_OFF = inverter.ComponentState.COMPONENT_STATE_SWITCHING_OFF
+    """The inverter is switching off and needs some time to fully shut down."""
+
+    STANDBY = inverter.ComponentState.COMPONENT_STATE_STANDBY
+    """The inverter is in a standby state, and is disconnected from the grid.
+
+    When connected to the grid, it run a few tests, and move to the `IDLE` state.
+    """
+
+    IDLE = inverter.ComponentState.COMPONENT_STATE_IDLE
+    """The inverter is idle."""
+
+    CHARGING = inverter.ComponentState.COMPONENT_STATE_CHARGING
+    """The inverter is consuming electrical energy to charge batteries.
+
+    Applicable to `BATTERY` and `HYBRID` inverters only.
+    """
+
+    DISCHARGING = inverter.ComponentState.COMPONENT_STATE_DISCHARGING
+    """The inverter is generating electrical energy."""
+
+    ERROR = inverter.ComponentState.COMPONENT_STATE_ERROR
+    """The inverter is in a faulty state."""
+
+    UNAVAILABLE = inverter.ComponentState.COMPONENT_STATE_UNAVAILABLE
+    """The inverter is online, but currently unavailable.
+
+    Possibly due to a pre- scheduled maintenance.
+    """
+
+    UNKNOWN = inverter.ComponentState.COMPONENT_STATE_UNKNOWN
+    """The state is provided by the device, but it is not one of the above states."""
```

Comment:
I think we should say component, not device.

```suggestion
    """A state is provided by the component, but it is not one of the above states."""

Reply:
This comes from the API specs, I guess we could change the wording here but not sure if we should _elevate_ it to the API specs. What do you think?